apiserver/facades/agent/uniter/uniter.go

```diff
@@ -2434,8 +2433,8 @@ func (u *UniterAPI) WatchConfigSettingsHash(ctx context.Context, args params.Ent
 // uniter can use the hash to determine whether the actual values have
 // changed since it last saw the config.
 func (u *UniterAPI) WatchTrustConfigSettingsHash(ctx context.Context, args params.Entities) (params.StringsWatchResults, error) {
-	getWatcher := func(unit *state.Unit) (state.StringsWatcher, error) {
-		return unit.WatchApplicationConfigSettingsHash()
+	getWatcher := func(ctx context.Context, unitName coreunit.Name) (watcher.StringsWatcher, error) {
```

Comment:
todo: it looks like the `getWatcher` methods are no longer required.  Just pass the `u.applicationService.Watch*Hash` methods to `watchHashes` now.

Reply:
This one is still needed because we get the application off the unit name, though one of them we use the unit name directly, and there we can delete this method. Have done that.